packages/@lwc/engine/src/framework/base-lightning-element.ts

```diff
@@ -287,6 +288,9 @@ function BaseLightningElementConstructor(this: LightningElement) {
         mode,
         delegatesFocus: !!ctor.delegatesFocus,
     };
+    if (useSyntheticShadow) {
+        assign(shadowRootOptions, { '$$lwc-synthetic-mode$$': true });
```

Comment:
I think it is safe to avoid the assign, just add it to the init obj in the line above to avoid the perf penalty